### ISSUE TITLE
add virtual crystal support for ml_norm

### ISSUE THREAD

**Issue by gefeichen** [UCL/STIR#691]
I add virtual crystal support for ml_norm 3d function. 
new_num_rings = num_rings - (num_axial_block-1) *num_axial_virtual_crystal
new_num_detector_perring = num_detector_perring - num_transaxial_block *num_transaxial_virtual_crystal

**Comment by KrisThielemans** [comment]
Hi. I guess you mean in `make_fan_data_remove_gaps` here
https://github.com/UCL/STIR/blob/c4df936a480c01441fb599ad7883a65a9566111c/src/buildblock/ML_norm.cxx#L1057-L1063
and using `Scanner::get_num_virtual_axial_crystals_per_block()` etc.

I believe that some of the other variables need changing as well, e.g.
```c++
const int new_fan_size = fan_size - num_transaxial_blocks_in_fansize*num_virtual_transaxial_crystals_per_block;
```
and something similar for the `max_delta` case.

Also
https://github.com/UCL/STIR/blob/c4df936a480c01441fb599ad7883a65a9566111c/src/buildblock/ML_norm.cxx#L1089-L1092

@tniknejad maybe you have some suggestions?

**Comment by KrisThielemans** [comment]
we could/should put some of this in `Scanner` probably. I thought `get_virtual_detection_pos(DetectionPosition&)` or so, but there's a bit of a nomenclature problem that the current `DetectionPosition` is always the "virtual" one.

**Comment by gefeichen** [comment]
Thanks for your reply. 
I tried to submit a commit with these modifications on apply_normfactors3D and find_normml3d (maybe failed). 

> https://github.com/UCL/STIR/blob/c4df936a480c01441fb599ad7883a65a9566111c/src/utilities/apply_normfactors3D.cxx#L78-L95


fan_size and max_delta should be modified on `make_fan_data_remove_gaps` if the virtual crystal is removed. And I don't know how to do it.

**Comment by KrisThielemans** [comment]
Thias was implemented in #833
